registry/storage/driver/s3/s3.go

```diff
@@ -724,3 +736,17 @@ func getPermissions() s3.ACL {
 func (d *driver) getContentType() string {
 	return "application/octet-stream"
 }
+
+// getbuf returns a buffer from the driver's pool with length d.ChunkSize. The
+// buffer provided is cleared of any data.
+func (d *driver) getbuf() []byte {
+	return d.pool.Get().([]byte)
+}
+
+func (d *driver) putbuf(p []byte) {
+	// clear before returning buffer
+	for i := 0; i < len(p); i++ {
```

Comment:
Why is this necessary, the buffer is always explicitly overwritten by read commands and the read length is checked.